deepchem/models/torch_models/layers.py

```diff
@@ -2682,3 +2682,336 @@ def forward(
+				 595-608.
+	"""
+
+  def __init__(self,
+               n_atom_input_feat: int = 75,
+               n_pair_input_feat: int = 14,
+               n_atom_output_feat: int = 50,
+               n_pair_output_feat: int = 50,
+               n_hidden_AA: int = 50,
+               n_hidden_PA: int = 50,
+               n_hidden_AP: int = 50,
+               n_hidden_PP: int = 50,
+               update_pair: bool = True,
+               init_: str = 'xavier_uniform_',
+               activation: str = 'relu',
+               batch_normalize: bool = True,
+               **kwargs):
+    """
+		Parameters
+		----------
+		n_atom_input_feat: int, optional (default 75)
+			Number of features for each atom in input.
+		n_pair_input_feat: int, optional (default 14)
+			Number of features for each pair of atoms in input.
+		n_atom_output_feat: int, optional (default 50)
+			Number of features for each atom in output.
+		n_pair_output_feat: int, optional (default 50)
+			Number of features for each pair of atoms in output.
+		n_hidden_AA: int, optional (default 50)
+			Number of units(convolution depths) in corresponding hidden layer
+		n_hidden_PA: int, optional (default 50)
+			Number of units(convolution depths) in corresponding hidden layer
+		n_hidden_AP: int, optional (default 50)
+			Number of units(convolution depths) in corresponding hidden layer
+		n_hidden_PP: int, optional (default 50)
+			Number of units(convolution depths) in corresponding hidden layer
+		update_pair: bool, optional (default True)
+			Whether to calculate for pair features,
+			could be turned off for last layer
+		init: str, optional (default 'xavier_uniform_')
+			Weight initialization for filters.
+		activation: str, optional (default 'relu')
+			Activation function applied
+		batch_normalize: bool, optional (default True)
+			If this is turned on, apply batch normalization before applying
+			activation functions on convolutional layers.
+		"""
+    super(WeaveLayer, self).__init__(**kwargs)
+    self.init: str = init_  # Set weight initialization
+    self.activation: str = activation  # Get activations
+    self.activation_fn: Callable = get_activation(activation)
+    self.update_pair: bool = update_pair  # last weave layer does not need to update
+    self.n_hidden_AA: int = n_hidden_AA
+    self.n_hidden_PA: int = n_hidden_PA
+    self.n_hidden_AP: int = n_hidden_AP
+    self.n_hidden_PP: int = n_hidden_PP
+    self.n_hidden_A: int = n_hidden_AA + n_hidden_PA
+    self.n_hidden_P: int = n_hidden_AP + n_hidden_PP
+    self.batch_normalize: bool = batch_normalize
+
+    self.n_atom_input_feat: int = n_atom_input_feat
+    self.n_pair_input_feat: int = n_pair_input_feat
+    self.n_atom_output_feat: int = n_atom_output_feat
+    self.n_pair_output_feat: int = n_pair_output_feat
+
+    #Construct internal trainable weights
+    init = getattr(initializers, self.init)
+    self.W_AA: torch.Tensor = init(
```

Comment:
explain in a comment about these variables, try to use better variable name if possible

Reply:
Yes, done